src/WorldMap.cpp

```diff
@@ -40,3 +235,200 @@ void WorldMap::clearMap()
 {
 	m_tileLayers.clear();
 }
+
+void WorldMap::addTileSheet(const std::string& tileSheetPath, const TileSheetDetails & tileSheetDetails)
+{
+	assert(!hasTileLayer(tileSheetDetails.m_name));
+	m_tileSheets.emplace(tileSheetDetails.m_name, TileSheet(tileSheetDetails, tileSheetPath));
+}
+
+void WorldMap::parseTileMap(const TiXmlElement & root, const LevelDetails & levelDetails)
+{
+	const int tileLayerCount = getNumberOfTileLayers(root);
```

Comment:
Pretty sure I left a comment on the last review about how this should be looped? No reason to iterate twice through the nodes like this.  Remove this getNumberOfTileLayers function, and instead of 'for (int i = 1, ...)' do something like:

for (const TiXmlElement* e = root.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
	if (e->Value() == std::string("layer"))

(I copied this from the getNumberOfTileLayers function).